pyk/src/pyk/integration_tests/test_proofs.py

```diff
@@ -42,3 +42,45 @@ def test_prove_claim_rule_profile(self):
+    KOMPILE_OUTPUT_DIR = 'definitions/imp'
+    KOMPILE_EMIT_JSON = True
+
+    KPROVE_USE_DIR = '.imp'
+
+    @staticmethod
+    def _update_symbol_table(symbol_table):
+        pass
+
+    def test_get_basic_block(self):
+        # Given
+        new_claim = KClaim(KToken('<k> $s = 0 ; while ( 0 <= $n ) { $s = $s + $n ; $n = $n + -1 ; } => . ... </k> <state> $n |-> (N => 0) $s |-> (_ => (N *Int (N +Int 1)) /Int 2) </state>', 'KCell'))
+
+        # When
+        post_depth_actual, post_branching_actual, post_state = self.kprove.get_claim_basic_block('imp-basic-block', new_claim)
+        post_state_pretty_actual = self.kprove.pretty_print(post_state)
+
+        post_depth_expected = 9
+        post_branching_expected = True
+        post_state_pretty_expected = '<generatedTop>\n' +                                                                                                            \
+                                     '  <T>\n' +                                                                                                                     \
+                                     '    <k>\n' +                                                                                                                   \
+                                     '      if ( 0 <=Int N ) { { $s = $s + $n ; $n = $n + -1 ; } while ( 0 <= $n ) { $s = $s + $n ; $n = $n + -1 ; } } else { }\n' + \
+                                     '      ~> _DotVar2\n' +                                                                                                         \
+                                     '    </k>\n' +                                                                                                                  \
+                                     '    <state>\n' +                                                                                                               \
+                                     '      $n |-> N $s |-> 0\n' +                                                                                                   \
+                                     '    </state>\n' +                                                                                                              \
+                                     '  </T>\n' +                                                                                                                    \
+                                     '  <generatedCounter>\n' +                                                                                                      \
+                                     '    _DotVar3\n' +                                                                                                              \
+                                     '  </generatedCounter>\n' +                                                                                                     \
+                                     '</generatedTop>'
```

Comment:
@nishantjr is this how you want it formatted?

Reply:
Yeah, but we don't need the `+`s or the line continuations I think.

Reply:
see here: https://github.com/runtimeverification/k/blob/master/pyk/src/pyk/tests/test_kcfg.py#L338

Reply:
```suggestion
        post_state_pretty_expected = ('<generatedTop>\n'
                                      '  <T>\n'
                                      '    <k>\n'
                                      '      if ( 0 <=Int N ) { { $s = $s + $n ; $n = $n + -1 ; } while ( 0 <= $n ) { $s = $s + $n ; $n = $n + -1 ; } } else { }\n'
                                      '      ~> _DotVar2\n'
                                      '    </k>\n'
                                      '    <state>\n'
                                      '      $n |-> N $s |-> 0\n'
                                      '    </state>\n'
                                      '  </T>\n'
                                      '  <generatedCounter>\n'
                                      '    _DotVar3\n'
                                      '  </generatedCounter>\n'
                                      '</generatedTop>')
```

I'm not sure if the parens are needed.

Reply:
The parens are needed, at least I couldn't fine a way without. Looks much nicer.